### DISCUSSION TITLE
Expose some functionality to global scope

### DISCUSSION THREAD

**Discussion by KevinBatdorf** [alpine-collective/alpine-magic-helpers#110] [Ideas]
From Discord:

> I might have an unpopular opinion regarding  some magic helpers. I think $fetch, $interval, $range, $screen, $scroll and $truncate would be more useful as stand-alone helpers outside the scope of Alpine. They don't hold any Alpine specific logic and yet they are scoped to be used only from within Alpine components. For example I was using $truncate but  realized that I also needed the functionality in some javascript outside my Alpine components and so I had to copy the code over to a generic global helper. Unless I am missing something, I don't see what these helpers bring to Alpine specifically or why they couldn't be used outside the scope of a component.

I tend to agree. We don't have to document it heavily (maybe just a footnote). 

Maybe `window.AlpineMagicHelpers.$post()`? 

I wonder if this would also work nicely:
```js
window.AMH.$component('my-component').myProp = false
```

**Comment by SimoTod** [comment]
Some of the helpers only make sense inside alpine like $parent, $refresh and in general anything that touches the alpine component passed into the function.
There are already plenty of options for doing those things outside alpine however you can expose it if you want. It kinda pollutes the global scope but it seems to be acceptable nowadays, hahaha.

**Reply by HugoDF** [reply]
Yeah was about to say, I remember people complaining they don't know how to do a HTTP GET in Alpine
